src/main/java/com/commercetools/sync/commons/helpers/BaseSyncStatistics.java

```diff
@@ -0,0 +1,237 @@
+    }
+
+    /**
+     * Builds a human readable processing time, as string, in the following format @{code "0d, 0h, 0m, 2s, 545ms"}
+     * and stores it in the publicly exposed variable {@code humanReadableProcessingTime}.
+     */
+    private void setHumanReadableProcessingTime() {
+        final long completeDaysInHours = TimeUnit.DAYS.toHours(processingTimeInDays);
+        final long completeHoursInMinutes = TimeUnit.HOURS.toMinutes(processingTimeInHours);
+        final long completeMinutesInSeconds = TimeUnit.MINUTES.toSeconds(processingTimeInMinutes);
+        final long completeSecondsInMillis = TimeUnit.SECONDS.toMillis(processingTimeInSeconds);
+
+        final long remainingHours = processingTimeInHours - completeDaysInHours;
+        final long remainingMinutes = processingTimeInMinutes - completeHoursInMinutes;
+        final long remainingSeconds = processingTimeInSeconds - completeMinutesInSeconds;
+        final long remainingMillis = processingTimeInMillis - completeSecondsInMillis;
+
+        humanReadableProcessingTime = format("%dd, %dh, %dm, %ds, %dms",
+                processingTimeInDays,
+                remainingHours,
+                remainingMinutes,
+                remainingSeconds,
+                remainingMillis
+        );
+    }
+
+    /**
+     * Gets the human readable processing time in the following format @{code "0d, 0h, 0m, 2s, 545ms"}.
+     *
+     * @return the human readable processing time in the following format @{code "0d, 0h, 0m, 2s, 545ms"}
+     */
+    public String getHumanReadableProcessingTime() {
+        return humanReadableProcessingTime;
+    }
+
+    /**
+     * Gets the number of days it took to process.
+     *
+     * @return number of days taken to process.
+     */
+    public long getProcessingTimeInDays() {
+        return processingTimeInDays;
+    }
+
+    /**
+     * Gets the number of hours it took to process.
+     *
+     * @return number of hours taken to process.
+     */
+    public long getProcessingTimeInHours() {
+        return processingTimeInHours;
+    }
+
+    /**
+     * Gets the number of minutes it took to process.
+     *
+     * @return number of minutes taken to process.
+     */
+    public long getProcessingTimeInMinutes() {
+        return processingTimeInMinutes;
+    }
+
+    /**
+     * Gets the number of seconds it took to process.
+     *
+     * @return number of seconds taken to process.
+     */
+    public long getProcessingTimeInSeconds() {
+        return processingTimeInSeconds;
+    }
+
+    /**
+     * Gets the number of milliseconds it took to process.
+     *
+     * @return number of milliseconds taken to process.
+     */
+    public long getProcessingTimeInMillis() {
+        return processingTimeInMillis;
+    }
+
+    /**
+     * Gets a summary message of the statistics report.
+     *
+     * @return a summary message of the statistics report.
+     */
+    public String getReportMessage() {
```

Comment:
This method returns empty String. Why don't implement it to return something reasonable or mark as abstract?